src/routes/subscriptions.rs

```diff
@@ -47,35 +49,131 @@ pub async fn subscribe(
         Err(_) => return StatusCode::BAD_REQUEST,
     };
 
-    match insert_subscriber(&app_state.pool, &new_subscriber).await {
-        Ok(_) => StatusCode::OK,
-        Err(_) => StatusCode::INTERNAL_SERVER_ERROR,
+    let mut transaction = match app_state.pool.begin().await {
```

Comment:
`sqlx` provides a `Transaction` struct that can wrap multiple queries into one transaction that either all succeeds or none of it does. This helps us since were performing multiple queries against multiple tables when this endpoint is called